### DISCUSSION TITLE
discussion about the reverse engineering of the protocol

### DISCUSSION THREAD

**Discussion by vbelloir** [custom-components/ble_monitor#219] [Q&A]
Hi team,
I don't know if this post is at the right place, because it is not an issue, just a question.
So just tell me if this is not the right place.

I have a question concerning this doc by @Ernst79 -thanks for that doc!)

https://github.com/custom-components/ble_monitor/blob/master/protocol.md

I really don't understand  len parts of advertising data.

There are 4 different lengths. I understand the two last (one for the sensors data length, 4 or 2, one for all between AD type and sensors data, included, 15 ou 13 bytes).

But I can find the meaning of both len value around the MAC Address. They are coded respectively on 7 and 4 bytes?

can you explain it to me?

Thanks for your help?
Vincent

**Comment by Ernst79** [comment]
The first Lenght (37) is the total length, starting from the first value after Len till the end (including RSSI) 
In the second Length, I made a small mistake (forgot to convert some of the values), I have changed it now.
This Length is the length from the first value after Len till the end minus one (so without RSSI)

**Reply by vbelloir** [reply]
Ok, I was misreading your table.

I thought that the len param was 04 3e 25 02 01 00 00 not only the third byte 0x25 (37 in decimal). I understand. 
So, do you know 04 3e means? and 02 01 00?

**Reply by Ernst79** [reply]
I don’t know, but I see 03 in the btmon output (LE Meta Event (0x3e)).
I also see 02 in the btmon output (LE Advertising Report (0x02))

**Reply by vbelloir** [reply]
ok, thanks.
